lib/modulesync/git.rb

```diff
@@ -21,9 +21,15 @@ def self.pull(git_base, name, branch, opts)
       else
```

Comment:
You still need the following at line 19 to deal with a freshly cloned repo:
 puts "Switching to branch #{branch}"
 repo.checkout(branch)